File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int res = 0;
        int l = 0, r = 1;
        while(r < prices.Length) {
            if(prices[l] > prices[r]) {
                l = r;
            }
            int check = prices[r] - prices[l];
            if(check > res) {
                res = check; 
            }
            r++;
        }
        return res;
    }
}

File: Data Structures & Algorithms/number-of-good-pairs/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    numIdenticalPairs(nums) {
        var res = 0;
        var tracker = new Map();
        for(var i = 0; i < nums.length; i++) {
            res += tracker.get(nums[i]) ?? 0;
            tracker.set(nums[i], (tracker.get(nums[i]) ?? 0) + 1);
        }
        return res;
    }
}
